command/lock.go

```diff
@@ -165,6 +170,12 @@ func (c *LockCommand) Run(args []string) int {
 	go func() {
 		if err := c.startChild(script, childDone, passStdin); err != nil {
 			c.Ui.Error(fmt.Sprintf("%s", err))
+
+			if c.childExitCode {
+				c.childLock.Lock()
```

Comment:
Hi @nicholascapo - I was wondering what you were using the lock for here? It doesn't seem like you need it.